### PR TITLE
Feat 71/create review

### PR DESCRIPTION
**Pull request by AKILBEK-UR**

### CODE REVIEW

File: src/modules/reviews/reviews.service.ts [hunk 1]
```diff
@@ -27,4 +38,38 @@ export class ReviewsService {
...
+    }
+
+    const user = await this.userRepository.findOne({
+      where: { id: userId },
+    });
+    if (!user) {
+      throw new NotFoundException('User not found with this id');
+    }
+
+    const userHasBooking = await this.bookingRepository
```
[the remark this answers is not in the document]
Comment (AKILBEK-UR) [reply src/modules/reviews/reviews.service.ts:58]: we cannot use it there

File: src/modules/reviews/reviews.service.ts [hunk 2]
```diff
@@ -27,4 +38,38 @@ export class ReviewsService {
...
+  async createReview(createReviewDto: CreateReviewDto, userId: string) {
+    const accommodation = await this.accommodationRepository.findOne({
+      where: { id: createReviewDto.accommodationId },
+    });
+
+    if (!accommodation) {
+      throw new NotFoundException('Accommodation not found with this id');
+    }
+
+    const user = await this.userRepository.findOne({
```
[the remark this answers is not in the document]
Comment (AKILBEK-UR) [reply src/modules/reviews/reviews.service.ts:51]: good comment

**Review by purple-mountain — APPROVED** [review]

File: src/modules/reviews/reviews.service.ts [hunk 3]
```diff
@@ -27,4 +38,27 @@ export class ReviewsService {
...
   }
+
+  async createReview(createReviewDto: CreateReviewDto, userId: string) {
+    const accommodation = await this.accommodatioService.getAccommodationById(
+      createReviewDto.accommodationId,
+    );
+
+    const user = await this.userService.getUser(userId);
+
+    const userBookings = await this.bookingService.getMyBookings({}, userId);
```
Comment (farrooh) [inline src/modules/reviews/reviews.service.ts:49]: You should check if user has completed (paid & checked out) booking.

**Review by farrooh — APPROVED** [review]
You can merge after resolving the comment.

**Pull request merged**